updated outdated dependencies

javabeanz opened the pull request:
updated outdated dependencies and
clean up poms with sortpom plugin

jtmelton commented:
@javabeanz just wanted to point out that this PR failed the build. Can you take a look and see what's happening? Also, this is a big PR due to reformatted files. I took a quick peek, and it looks like there's no content change in many of the files. Can you please re-work the PR with just the relevant changes so it's a bit easier to review?

javabeanz commented:
apparently there is an enforcer rule about junit 4.11. fixed in next commit

javabeanz commented:
not sure why junit has  both 2 versions ${junit.version} and hardcoded 4.11

jtmelton commented:
@javabeanz ok, looks like the build passes now - thanks! However, I still can't tell what changes were made because all the poms are completely re-organized. I do see the dependency status addition to the readme, but beyond that I don't know what else has changed. Can you either clean up this one or re-submit a new PR with only changes? I'm guessing all these files changed because of your XML editor settings in the IDE, but it makes review very difficult.

jtmelton commented:
@javabeanz ok, looked at this again, and by the link in the readme, I think I get what you're after here. I'm closing this issue since the changes are significant and it's very hard to review. Also, the readme link points to your branch, not master. If you want to let me know what steps you went through to get this going and/or submit another PR, I'm happy to accept as long as the relevant changes are obvious and the link is to the master branch.

javabeanz commented:
wow this is disappointing.

jtmelton commented:
Ok re-opening to make it clear this PR is very valuable to us, but I still can't review all these as I can't tell what the actual changes are since _everything_ in the file has changed. If you can address the following 2 issues, I'm happy to merge: 

- re-format the XML files in our format so the changes in version number become obvious and it's easier to review
- change the link on the readme to the actual project master instead of your branch

javabeanz commented:
Given that all test succeed, I thought it would be a simple merge. If it
doesn't work that way, why do we have those tests ? Or maybe there aren't
enough tests yet ? then we should work on that. Where I worked in teams,
unit tests are meant to safeguard against refactor mistakes.

I used the sortpom maven plugin to make the poms more standard format :
https://github.com/Ekryd/sortpom.

The free Versioneye service needs OAuth access to your github accoun, this
is not something I can give, so that link I cannot change to your branch.

I hope this addresses your issues, but sense that you are used to work
differently.

Kind regards,
sytze.


On Tue, Jul 19, 2016 at 6:04 PM, John Melton <notifications@github.com>
wrote:

> Ok re-opening to make it clear this PR is very valuable to us, but I still
> can't review all these as I can't tell what the actual changes are since
> *everything* in the file has changed. If you can address the following 2
> issues, I'm happy to merge:
>
>    - re-format the XML files in our format so the changes in version
>    number become obvious and it's easier to review
>    - change the link on the readme to the actual project master instead
>    of your branch
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jtmelton/appsensor/pull/50#issuecomment-233681831>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADE4rLUJgj3Mh49GTSj7cCQHmNdtnnoJks5qXPWhgaJpZM4JD9_a>
> .
>

jtmelton commented:
Ok, thanks for the feedback. Clearly I did not do well communicating the issue. I fully believe this change will NOT break anything. I have 2 requests: 

1. Code formatting - my request around the XML files is not because I think you broke anything. I suspect you did not because as you said, the tests pass. It's just a matter of formatting. I'd also like to understand what libraries actually changed, so that on the off chance it messes something up in a user environment (library version mismatch) then the diff is clear on what changed. In general I try to make diffs as minimal as possible. If we want to do a formatting change, then that should be the _only_ thing that changes. We should not change the formatting _and_ versions at the same time, since that will make it hard to understand what caused the breakage in the case that anything does break. 

2. The versioneye service pointer. I understand what you mean now about the OAuth access, and I can fix that on my own. My only remaining point now is around point 1. 

Thanks,
John

javabeanz commented:
I am looking for ways to do this efficiently, maybe with a new branch ? Is
it an option for you to run the sortpom plugin on your master and compare
my changes against the sorted poms ? If there is a simple way for me to do
this, I would do it, but I dont know yet how.

On Mon, Jul 25, 2016 at 7:30 PM, John Melton <notifications@github.com>
wrote:

> Ok, thanks for the feedback. Clearly I did not do well communicating the
> issue. I fully believe this change will NOT break anything. I have 2
> requests:
>
>    1.
>
>    Code formatting - my request around the XML files is not because I
>    think you broke anything. I suspect you did not because as you said, the
>    tests pass. It's just a matter of formatting. I'd also like to understand
>    what libraries actually changed, so that on the off chance it messes
>    something up in a user environment (library version mismatch) then the diff
>    is clear on what changed. In general I try to make diffs as minimal as
>    possible. If we want to do a formatting change, then that should be the
>    *only* thing that changes. We should not change the formatting *and*
>    versions at the same time, since that will make it hard to understand what
>    caused the breakage in the case that anything does break.
>    2.
>
>    The versioneye service pointer. I understand what you mean now about
>    the OAuth access, and I can fix that on my own. My only remaining point now
>    is around point 1.
>
> Thanks,
> John
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jtmelton/appsensor/pull/50#issuecomment-235023518>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ADE4rPV6uQzdkilnc2alax6JT2I65dBAks5qZPKugaJpZM4JD9_a>
> .
>

jtmelton commented:
I think the easiest way (if it works) is to change your formatting settings for XML to match those already used on the project. You can see what I've got here - 

![image](https://cloud.githubusercontent.com/assets/1624464/17113653/abefa2ea-5279-11e6-924c-5e4018821b33.png)

If that doesn't work, then you might either try a new branch or even just telling me how to do it. I'd prefer if you are able to submit a PR, though, so you can get the credit since you've done the work.
